### DISCUSSION TITLE
ActiveModel and NOT NULL

### DISCUSSION THREAD

**Discussion by cemoktra** [SeaQL/sea-orm#465] [Q&A] [answered]
Hi,

a question about inserting today. `ActiveModel` implements `Default` which is fine, but might allow trying to insert something that has fields defaulted to `NotSet` ending up in an database error.

Not sure whats the best solution here. Maybe a function to test insert-ability against the schema or an `InsertModel` that has those `NOT NULL` fields as mandatory instead of optional.

**Comment by billy1624** [comment, marked as answer]
Hey @cemoktra, thanks for raising this issue. Yes, we notice this problem. We are discussing to introduce `InsertModel`, just like what you have suggested. Related issue #101. Welcome to discuss here :)
